src/Runner.Listener/BrokerMessageListener.cs

```diff
@@ -108,7 +108,7 @@ public async Task<TaskAgentMessage> GetNextMessageAsync(CancellationToken token)
 
                     if (!IsGetNextMessageExceptionRetriable(ex))
                     {
-                        throw;
+                        throw new NonRetryableException("Get next message failed with non-retryable error.", ex);
```

Comment:
We didn't do this with the other Listener since we are preventing the loop during CreateSeesion.

Since the broker listener doesn't really have a concept for create session, i need to break out the loop at here for now.